src/modules/iotjs_module_buffer.c

```diff
@@ -252,6 +252,17 @@ JHANDLER_FUNCTION(Compare) {
 }
 
 
+#define CONVERT_DOUBLE_TO_SIZE(n, d) \
+  size_t n;                          \
+  do {                               \
+    if (d < 0) {                     \
```

Comment:
@LaszloLango Basically I agree with your suggestion. However, I've checked how node.js deal with this situation.

First, I searched Buffer API documentation in Node. It say nothing about NaN, +Inf/-Inf except for `lastIndexOf`, 'IndexOf'. Second, I've done some tests with Node.js. It seemingly work differently for each APIs.

```
> new Buffer(Infinity)
RangeError: Invalid typed array length
    at new Uint8Array (native)
    at createBuffer (buffer.js:66:15)
    at allocate (buffer.js:183:12)
    at Function.Buffer.allocUnsafe (buffer.js:151:10)
    at new Buffer (buffer.js:93:19)
    at repl:1:1
    at REPLServer.defaultEval (repl.js:269:27)
    at bound (domain.js:287:14)
    at REPLServer.runBound [as eval] (domain.js:300:12)
    at REPLServer.<anonymous> (repl.js:439:12)
> new Buffer(-Infinity)
<Buffer >
> new Buffer(NaN)
<Buffer >
> new Buffer(0)
<Buffer >
> var b = new Buffer(8)
undefined
> console.log(b)
<Buffer 78 ee bc c2 83 7f 00 00>
undefined
> var c = new Buffer(8)
undefined
> console.log(c)
<Buffer 80 a5 0c 03 00 00 00 00>
undefined
> b.copy(c, 0, 0)
8
> console.log(c)
<Buffer 78 ee bc c2 83 7f 00 00>
undefined
> b.copy(c, 0, NaN)
8
> c = new Buffer(8)
<Buffer 80 a5 0c 03 00 00 00 00>
> b.copy(c, 0, NaN)
8
> console.log(c)
<Buffer 78 ee bc c2 83 7f 00 00>
undefined
> c = new Buffer(8)
<Buffer 0a 00 00 00 00 00 00 00>
> b.copy(c, 0, -Infinity)
8
> console.log(c)
<Buffer 78 ee bc c2 83 7f 00 00>
undefined
> c = new Buffer(8)
<Buffer 08 a6 0c 03 00 00 00 00>
> b.copy(c, 0, 0, 2)
2
> console.log(c)
<Buffer 78 ee 0c 03 00 00 00 00>
undefined
> b.copy(c, 0, 0, Infinity)
0
> console.log(c)
<Buffer 78 ee 0c 03 00 00 00 00>
undefined
> b.copy(c, 0, Infinity)
8
> console.log(c)
<Buffer 78 ee bc c2 83 7f 00 00>
```

I suggest `0`, `0`, `SIZE_MAX` for NaN, -inf and inf respectively. I've changed your suggestion slightly because `CONVERT_DOUBLE_TO_SIZE` returns `size_t`.